crates/oxc_diagnostics/src/service.rs

```diff
@@ -202,33 +205,28 @@ impl DiagnosticService {
                     }
                 }
 
-                if self.silent {
+                if self.silent || is_minified_file {
                     continue;
                 }
 
                 if let Some(err_str) = self.reporter.render_error(diagnostic) {
                     // Skip large output and print only once.
                     // Setting to 1200 because graphical output may contain ansi escape codes and other decorations.
                     if err_str.lines().any(|line| line.len() >= 1200) {
+                        is_minified_file = true;
                         let minified_diagnostic = Error::new(
                             OxcDiagnostic::warn("File is too long to fit on the screen").with_help(
                                 format!("{} seems like a minified file", path.display()),
                             ),
                         );
 
                         if let Some(err_str) = self.reporter.render_error(minified_diagnostic) {
-                            writer
-                                .write_all(err_str.as_bytes())
-                                .or_else(Self::check_for_writer_error)
-                                .unwrap();
+                            output = err_str;
                         }
-                        break;
+                        continue;
```

Comment:
This may not be ideal, let me think about this.

Reply:
we're always going to need to loop over diagnostics to get the count. but we could do something like check the first diagnostic's src code length to guess whether it is minified or not